### PR TITLE
pipr: init at 0.0.12

### PR DESCRIPTION
**Pull request by elkowar**
###### Motivation for this change
https://github.com/elkowar/pipr

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2448,6 +2448,12 @@
     githubId = 97852;
     name = "Ellis Whitehead";
   };
+  elkowar = {
```
Comment (mweinelt) [inline maintainers/maintainer-list.nix:2451]: Please move this change into a separate commit.

File: pkgs/applications/misc/pipr/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{ stdenv
+, fetchFromGitHub
+, rustPlatform
+, bubblewrap
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "pipr";
+  version = "v0.0.12";
```
Comment (mweinelt) [inline pkgs/applications/misc/pipr/default.nix:9]:
```suggestion
  version = "0.0.12";
```

File: pkgs/applications/misc/pipr/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = fetchFromGitHub {
+    owner = "ElKowar";
+    repo = pname;
+    rev = "v0.0.12";
```
Comment (mweinelt) [inline pkgs/applications/misc/pipr/default.nix:14]:
```suggestion
    rev = "v${version}";
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -22758,6 +22758,8 @@ in
 
   synergyWithoutGUI = synergy.override { withGUI = false; };
 
+  pipr = callPackage ../applications/misc/pipr { };
```
Comment (mweinelt) [inline pkgs/top-level/all-packages.nix:22761]: The file is somehwat sorted alphabetically, can you please try and find a better spot to put this?

File: pkgs/applications/misc/pipr/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+  cargoSha256 = "1gbzi4bx6chf8hpxbf99bnncdgi2xqj41fh9jldwz53rnfvwvayn";
+
+  buildInputs = [ bubblewrap ];
+
+  meta = with stdenv.lib; {
+    description = "A commandline-tool to interactively write shell pipelines";
+    homepage = "https://github.com/ElKowar/pipr";
+    license = licenses.mit;
+    maintainers = maintainers.elkowar;
```
Comment (mweinelt) [inline pkgs/applications/misc/pipr/default.nix:26]:
```suggestion
    maintainers = with maintainers; [ elkowar ];
```

This should be a list.

File: pkgs/applications/misc/pipr/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+, fetchFromGitHub
+, rustPlatform
+, bubblewrap
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "pipr";
+  version = "v0.0.12";
+
+  src = fetchFromGitHub {
```
Comment (mweinelt) [inline pkgs/applications/misc/pipr/default.nix:11]: I think it would be better if we used the upstream, not the mirror, don't you? `fetchGit` + https://gitlab.com/Elkowar/pipr

**Comment by mweinelt** [comment]
I'm getting the following error, so something with bubblewrap is not working out.
```sh
[nix-shell:~/.cache/nixpkgs-review/pr-94810]$ ./results/pipr/bin/pipr
bubblewrap installation not found. Please make sure you have `bwrap` on your path, or supply --no-isolation to disable safe-mode
```

Result of `nixpkgs-review pr 94810` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- pipr
</details>

**Review by mweinelt — CHANGES_REQUESTED** [review]
Thanks for your submission, I've added a few remarks.

File: pkgs/applications/misc/pipr/default.nix (hunk 6 above)
Comment (elkowar) [reply pkgs/applications/misc/pipr/default.nix:11]: generally yes. In this case, I'm planning on moving my projects over to github completely in the near future, so i makes sense to directly use github. If using a mirror in general is an issue, I can use gitlab here of course

Comment (mweinelt) [reply pkgs/applications/misc/pipr/default.nix:11]: Whatever works for you.

**Comment by elkowar** [comment]
I updated the PR, everything you mentioned should be fixed now, @mweinelt 
The documentation regarding how to contribute to nixpkgs mentions that the all-packages.nix file is sorted by category and _then_ alphabetically. If, apparently, this has changed to now just be fully sorted alphabetically, the documentation should probably be updated in that regard

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 94810` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- pipr
</details>

**Review by mweinelt — APPROVED** [review]

**Comment by mweinelt** [comment]
Sorry, took me a while to come back here.

**Comment by elkowar** [comment]
no worries, thanks!
